### PR TITLE
initial jack-audio support

### PR DESCRIPTION
**Pull request by lyrra**
Jack-audio support.
 Enabled at build-time by selecting MUE_ENABLE_AUDIO_JACK=ON, ie no run-time selection support between jack/alsa.

- [x ] I signed the [CLA](https://musescore.org/en/cla)

TODO:
- ~~hangs jackd (which cant recover even after musescore is killed)~~ (can't name client "default")
-  ~~outputs audio to jack, but it is butched~~ (was counting frames vs bytes)

Try it out:
1) start jackd with MuseScores default 44100Hz:
    ```jackd -r -d dummy -r 44100```
    (If using the dummy driver, record the audio using jack_capture).
2) start MuseScore
3) jack-connect musescore to jack's default output
   This is heusteristics that Mu3 did, but maybe should be handled by environment/setup managers such as Carla.
    ```
    $ jack_connect MuseScore:audio_out_right system:playback_2
    $ jack_connect MuseScore:audio_out_left system:playback_1
    ```

### CODE REVIEW

File: src/app/CMakeLists.txt [hunk 1]
```diff
@@ -207,6 +207,10 @@ if (MUE_BUILD_MUSESAMPLER_MODULE)
     list(APPEND LINK_LIB musesampler)
 endif(MUE_BUILD_MUSESAMPLER_MODULE)
 
+if (MUE_ENABLE_AUDIO_JACK)
+    list(APPEND LINK_LIB jack)
+endif(MUE_ENABLE_AUDIO_JACK)
```
Comment (igorkorsukov) [inline src/app/CMakeLists.txt:212]: Better move this to audio module cmake file, like https://github.com/musescore/MuseScore/blob/master/src/framework/audio/CMakeLists.txt#L273

Comment (igorkorsukov) [reply src/app/CMakeLists.txt:212]: I see that jack is already being linked, so it's probably not needed here

```
set(MODULE_LINK ${MODULE_LINK} ${JACK_LIBRARIES} pthread )
```

File: src/framework/audio/internal/platform/jack/jackaudiodriver.cpp [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+    jack_set_process_callback(handle, jack_process_callback, (void*)jackParam);
+
+    if (jack_activate(handle)) {
+        LOGE() << "cannot activate client";
+        return nullptr;
+    }
+
+    while (1) {
+        std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+    }
```
Comment (igorkorsukov) [inline src/framework/audio/internal/platform/jack/jackaudiodriver.cpp:74]: Maybe it's not done yet... probably there shouldn't be an endless empty loop with no way to exit it

Comment (lyrra) [reply src/framework/audio/internal/platform/jack/jackaudiodriver.cpp]: Thanks! fixed in a cleanup commit.

**Review by igorkorsukov — APPROVED** [review]

**Comment by igorkorsukov** [comment]
@lyrra Now the choice of the driver is made at the compilation stage, respectively, Jack support will not be included in the release builds.
Do you plan to add driver selection at runtime, to the settings?

**Comment by lyrra** [comment]
(squashed commits for cleaner history).

No plans so far on adding runtime selection support between jack and alsa.
Because that requires more in-depth understanding and perhaps UI-design.

**Pull request merged**